File: apps/web/src/lib/document-status.ts
export type DocumentWorkflowStatus =
  | "draft"
  | "sent"
  | "in_progress"
  | "waiting_for_payment"
  | "completed"
  | "cancelled"
  | "declined"
  | "expired";

const WORKFLOW_STATUSES: readonly DocumentWorkflowStatus[] = [
  "draft",
  "sent",
  "in_progress",
  "waiting_for_payment",
  "completed",
  "cancelled",
  "declined",
  "expired",
] as const;

export function isWorkflowStatus(
  status: string
): status is DocumentWorkflowStatus {
  return WORKFLOW_STATUSES.some((valid) => valid === status);
}

export function toWorkflowStatus(status: string): DocumentWorkflowStatus {
  return isWorkflowStatus(status) ? status : "draft";
}

File: packages/sdk-python/seal_api_client/models/document_status.py
from enum import StrEnum


class DocumentStatus(StrEnum):
    CANCELLED = "cancelled"
    COMPLETED = "completed"
    DECLINED = "declined"
    DRAFT = "draft"
    IN_PROGRESS = "in_progress"
    SENT = "sent"

    def __str__(self) -> str:
        return str(self.value)
